internal/reconciler/telemetry/log_components_checker.go

```diff
@@ -30,96 +30,100 @@ func (l *logComponentsChecker) Check(ctx context.Context, telemetryInDeletion bo
 		return &metav1.Condition{}, fmt.Errorf("failed to get list of LogParsers: %w", err)
 	}
 
-	reason := l.determineReason(logPipelines.Items, logParsers.Items, telemetryInDeletion)
-	status := l.determineConditionStatus(reason)
-	message := l.createMessageForReason(logPipelines.Items, logParsers.Items, reason)
-
-	conditionType := conditions.TypeLogComponentsHealthy
-	return &metav1.Condition{
-		Type:    conditionType,
-		Status:  status,
-		Reason:  reason,
-		Message: message,
-	}, nil
-}
+	if result := l.checkForResourseBlocksDeletionCondition(logPipelines.Items, logParsers.Items, telemetryInDeletion); result != nil {
+		return result, nil
+	}
 
-func (l *logComponentsChecker) determineReason(pipelines []telemetryv1alpha1.LogPipeline, parsers []telemetryv1alpha1.LogParser, telemetryInDeletion bool) string {
-	if telemetryInDeletion && (len(pipelines) != 0 || len(parsers) != 0) {
-		return conditions.ReasonResourceBlocksDeletion
+	if result := l.checkForNoPipelineDeployedCondition(logPipelines.Items); result != nil {
+		return result, nil
 	}
 
-	if len(pipelines) == 0 {
-		return conditions.ReasonNoPipelineDeployed
+	if result := l.checkForFirstUnhealthyPipelineCondition(logPipelines.Items); result != nil {
```

Comment:
why do we not present all those conditions to the user?

Reply:
It is a valid point that we maybe should not just copy over the first unhealthy reason but use an aggregating reason which points to the pipeline.. Still that should not be a concern of this PR